common/src/asset.rs

```diff
@@ -159,10 +232,49 @@ impl<T: AssetClass> Display for FungibleAsset<T> {
     }
 }
 
+impl<T: AssetClass> std::str::FromStr for FungibleAsset<T> {
+    type Err = FungibleAssetParseError;
+
+    fn from_str(s: &str) -> Result<Self, Self::Err> {
+        let parts: Vec<&str> = s.split(':').collect();
+
+        match parts.as_slice() {
+            ["nep141", contract_id] => {
+                let account_id = contract_id
+                    .parse::<AccountId>()
+                    .map_err(|e| FungibleAssetParseError::InvalidAccountId(e.to_string()))?;
+                Ok(FungibleAsset::nep141(account_id))
+            }
+            ["nep245", contract_id, token_id] => {
+                let account_id = contract_id
+                    .parse::<AccountId>()
+                    .map_err(|e| FungibleAssetParseError::InvalidAccountId(e.to_string()))?;
+
+                if token_id.is_empty() {
+                    return Err(FungibleAssetParseError::EmptyTokenId);
+                }
+
+                Ok(FungibleAsset::nep245(account_id, (*token_id).to_string()))
+            }
+            _ => Err(FungibleAssetParseError::InvalidFormat),
+        }
+    }
+}
+
+#[derive(Debug, thiserror::Error)]
+pub enum FungibleAssetParseError {
+    #[error("Invalid format. Expected 'nep141:contract_id' or 'nep245:contract_id:token_id'")]
```

Comment:
Might be better to clarify which tokens are literals and which are placeholders:

```suggestion
    #[error("Invalid format. Expected 'nep141:<contract_id>' or 'nep245:<contract_id>:<token_id>'")]
```

Reply:
Fixed